src/7-to-8/major-changes/suicide-triggers.rst

```diff
@@ -1,64 +1,57 @@
-.. _728.suicide_triggers:
+.. TODO: rename to "Graph branching, optional outputs and suicide triggers", and give example (the one I posted on Element)
 
-Suicide Triggers
-================
+.. _728.optional_outputs:
 
-.. admonition:: Does This Change Affect Me?
-   :class: tip
-
-   Suicide triggers are written using an exclamation mark ``!`` like so:
-
-   .. code-block:: cylc-graph
-
-      task_1 => ! task_2
+Graph branching, optional outputs and suicide triggers
+======================================================
 
-   Read this section if your workflows contain suicide triggers.
+Cylc 8 has a :ref:`new scheduling algorithm <728.scheduling_algorithm>` and
+a new syntax for dealing with tasks that may not necessarily complete.
+It handles :term:`graphs <graph>` in an event-driven manner which means
+that a workflow can follow different paths in different eventualities (without
+the need for suicide triggers). This is called :term:`graph branching`.
 
-Overview
---------
+.. admonition:: Does This Change Affect Me?
+   :class: tip
 
-In Cylc 7 :term:`suicide triggers <suicide trigger>` are used to remove
-tasks from the graph automatically at runtime.
+   This change affects you if you are upgrading a Cylc 7 workflow that
+   contains graph branches that are not necessarily expected to complete
+   at runtime. You might get a ``GraphParseError`` during validation with
+   Cylc 8.
 
-Cylc 8 handles :term:`graphs <graph>` in an event-driven manner which means
-that a workflow can follow different paths in different eventualities without
-the need for suicide triggers. This is called :term:`graph branching`.
+   Typically this will be the case if you are using any of the following in
+   the workflow graph:
 
-.. seealso::
+   - :term:`suicide triggers <suicide trigger>`
+   - the ``|`` (OR) operator
+   - the ``:fail`` output or custom ouputs
 
-   :ref:`Graph Branching` in the user guide.
+   You should *not* perform this upgrade if still in :ref:`cylc_7_compat_mode`
+   (``suite.rc`` filename).
 
 
-Changes
--------
-
-No changes are required, suicide triggers are still supported in Cylc 8,
-however, they are most likely unnecessary.
+Required Changes
+^^^^^^^^^^^^^^^^
 
+Any :term:`task outputs <task output>` that are not necessarily expected to
+complete must be marked as :term:`optional <optional output>` using ``?``.
+Suicide triggers can then be removed.
 
 Example
--------
+^^^^^^^
 
 Here is an example Cylc 7 :term:`graph`:
 
 .. code-block:: cylc-graph
 
-   # The success case.
-   foo => bar
+   foo:fail => recover
 
-   # The fail case.
-   bar:fail => recover
+   foo | recover => bar
 
    # Remove the "recover" task in the success case.
-   bar => ! recover
-
-   # Remove the "bar" task in the fail case.
-   recover => ! bar
-
-   # Downstream dependencies.
-   bar | recover => baz
-
-Which results in the following logic:
+   foo => ! recover
+   # Remove the "foo" task in the fail case.
+   recover => ! foo
```

Comment:
I have taken out the first task because it was unnecessary for the example.

That first edge was confusingly commented as the success case, but it wasn't really, the `bar => baz` was the success case

```mermaid
graph TD
    subgraph before
    A(foo) -->|unnecessary| B(bar) -->|success case| C(baz)
    B -.->|fail case| D(recover) -.-> C
    end
```
```mermaid
graph TD
    subgraph after
    A(foo) --> B(bar)
    A -.-> C(recover) -.->B
    end
```

What part of it would you say was clearer before?

Reply:
Yeah the first task was unnecessary. It was probably abstracted from an old example that was meant to look more realistic.